nixos/modules/programs/qdmr.nix

```diff
@@ -0,0 +1,27 @@
+{ config, lib, pkgs, ... }:
+
+{
+  meta.maintainers = [ lib.maintainers.janik ];
+
+  ###### interface
+  options = {
+    programs.qdmr = {
+      enable = lib.mkOption {
+        default = false;
+        type = lib.types.bool;
+        description = lib.mdDoc ''
+          Whether to configure system to use qdmr.
+          To grant access to a user, it must be part of dialout group:
+          `users.users.alice.extraGroups = ["dialout"];`
+        '';
+      };
+    };
+  };
+
+  ###### implementation
+  config = lib.mkIf config.programs.qdmr.enable {
+    services.udev.packages = [ pkgs.qdmr ];
+    environment.systemPackages = [ pkgs.qdmr ];
+    #users.groups.dialout = {};
```

Comment:
Why?

Reply:
What are you asking?
If you are wondering why it enables the package twice:
```
    services.udev.packages = [ pkgs.qdmr ];
    environment.systemPackages = [ pkgs.qdmr ];
```
then the answer is to get udev rules and have the package enabled, if you just activate the package with ``` environment.systemPackages = [ pkgs.qdmr ]; ``` the package works but it's missing the udev rules and you need those because this is a program to flash dmr-radios over usb. If you only activate the ``` services.udev.packages = [ pkgs.qdmr ]; ``` you won't get the bins, .desktop file, icons and anything else the program needs to work copied.
Thats the whole point of having a module as wrapper
The user needs to add them self to the dialout group for the udev rules to apply that's why there is the commented out line.

Reply:
Why your code is commented here? 

```nix
#it looks like [ you forgot something ]
```

Reply:
Ah you right i'm sorry ^^ fixed it